rotkehlchen/chain/ethereum/modules/eth2/beacon.py

```diff
@@ -55,31 +55,48 @@ def set_rpc_endpoint(self, rpc_endpoint: str) -> None:
         self.rpc_endpoint = rpc_endpoint.rstrip('/')
         result = self.query('eth/v1/node/version')
         try:
-            version = result['version']  # type: ignore  # related to the overload problem below
+            version = result['version']
         except KeyError as e:
             raise RemoteError(f'Failed to parse the node version response from {rpc_endpoint}') from e  # noqa: E501
 
         log.info(f'Connected to {rpc_endpoint} with beacon node {version}')
 
     @overload
-    def query(self, querystr: Literal['/eth/v1/node/version']) -> dict:  # type: ignore  # not sure how to fix the overlapping type at overload here
+    def query(  # type: ignore  # not sure how to fix the overlapping type at overload here
+            self,
+            endpoint: Literal['eth/v1/node/version'],
+            data: dict[str, Any] | None = None,
+    ) -> dict:
+        ...
+
+    @overload
+    def query(
+            self,
+            endpoint: Literal['eth/v1/beacon/states/head/validators'],
+            data: dict[str, Any],
+    ) -> list[dict]:
         ...
 
     @overload
-    def query(self, querystr: str) -> list[dict]:
+    def query(self, endpoint: str, data: dict[str, Any] | None = None) -> list[dict]:
         ...
 
-    def query(self, querystr: str) -> list[dict] | dict:
+    def query(self, endpoint: str, data: dict[str, Any] | None = None) -> dict | list[dict]:
         """
         May raise:
         - RemoteError due to problems querying the node
         """
-        querystr = self.rpc_endpoint + '/' + querystr
-        log.debug(f'Querying beacon node {querystr}')
+        url = self.rpc_endpoint + '/' + endpoint
+        log.debug(f'Querying beacon node {endpoint} with {data=}')
         try:
-            response = self.session.get(querystr, timeout=CachedSettings().get_timeout_tuple())
+            response = self.session.request(
+                url=url,
+                json=data,
+                method='POST' if endpoint == 'eth/v1/beacon/states/head/validators' else 'GET',
```

Comment:
why not make the method an argument of the function? This way you don't need this if everytime something is queried.